### ISSUE TITLE
windowFunnel: misleading description of window

### ISSUE THREAD

**Issue by jinghancc** [ClickHouse/ClickHouse#21849]
(you don't have to strictly follow this form)

**Describe the issue**
Documentation of `windowFunnel` describes parameter `window` as time interval between two consecutive conditions:
![image](https://user-images.githubusercontent.com/39145303/111563091-3eb71580-87d2-11eb-95e0-7fdfeaafed13.png)

but actually, it is the time interval between the first condition and last condition, so it should be:
```
timestamp of cond1 <= timestamp of cond2 <= ... <= timestamp of condN <= timestamp of cond1 + window.
```
It would be nice if we could correct this description since it's misleading to users.
**Additional context**

**Comment by sundy-li** [comment]
Not yet, the doc is right. The window is between sequence events. It's not between the first condition and last condition.

**Comment by jinghancc** [comment]
> Not yet, the doc is right. The window is between sequence events. It's not between the first condition and last condition.

Hi @sundy-li , thanks for your response, but the window is between the first condition and last condition based on test query.

First example: user 123 completed event 1, event 2, event 3 within time window 50, event chain is 130 -> 150 -> 180.
```sql
SELECT
    uid,
    windowFunnel(50)(event_ms, event_id = 1, event_id = 2, event_id = 3) AS level
FROM
    (
    /* emulate test dataset */
    SELECT data.1 event_id, data.2  event_ms, data.3 uid
    FROM (
        SELECT arrayJoin(
        [(1, 130, 123),  
        (2, 150, 123), 
        (3, 180, 123)]) data)
    )
WHERE
    event_ms >= 0
    AND event_ms <= 300
GROUP BY
    uid;

┌─uid─┬─level─┐
│ 123 │     3 │
└─────┴───────┘
```

Second example: user 123 only completed event 1 and event 2, because event 3 happened at 181, which is greater than 130 + 50.
```sql
SELECT
    uid,
    windowFunnel(50)(event_ms, event_id = 1, event_id = 2, event_id = 3) AS level
FROM
    (
    /* emulate test dataset */
    SELECT data.1 event_id, data.2  event_ms, data.3 uid
    FROM (
        SELECT arrayJoin(
        [(1, 130, 123),  
        (2, 150, 123), 
        (3, 181, 123)]) data)
    )
WHERE
    event_ms >= 0
    AND event_ms <= 300
GROUP BY
    uid;

┌─uid─┬─level─┐
│ 123 │     2 │
└─────┴───────┘
```

If the window here means `window between sequence events`, then both should generate results as 3.

We could also take a look at source code [here](https://github.com/ClickHouse/ClickHouse/blob/master/src/AggregateFunctions/AggregateFunctionWindowFunnel.h#L201), it shows that if current event satisfy the condition and is within the time window, current event always inherit the timestamp from previous event as its new timestamp, which means the window compares the first condition and last condition.

**Comment by sundy-li** [comment]
@jinghancc  Yes, you are right, I contributed these codes 3 years ago and I almost forgot that.

**Comment by jinghancc** [comment]
> @jinghancc Yes, you are right, I contributed these codes 3 years ago and I almost forgot that. Could you send a pr to improve that?

Hi @sundy-li , for sure, I will send a PR soon, thank you for contribution again!

**Comment by jinghancc** [comment]
> @jinghancc Yes, you are right, I contributed these codes 3 years ago and I almost forgot that. Could you send a pr to improve that?

@sundy-li by the way, I have another question about `windowFunnel` which is described [here](https://github.com/ClickHouse/ClickHouse/issues/21835), do you mind taking a look when you have time?

**Comment by DanRoscigno** [comment]
closed by #21901
